contributors/devel/sig-architecture/api-conventions.md

```diff
@@ -308,17 +308,49 @@ response reduces the complexity of these clients.
 
 ##### Typical status properties
 
-**Conditions** represent the latest available observations of an object's
-state.  They are an extension mechanism intended to be used when the details of
-an observation are not a priori known or would not apply to all instances of a
-given Kind.  For observations that are well known and apply to all instances, a
-regular field is preferred.  An example of a Condition that probably should
-have been a regular field is Pod's "Ready" condition - it is managed by core
-controllers, it is well understood, and it applies to all Pods.
+**Conditions** provide a standard mechanism for higher-level status reporting
+from a controller. They are an extension mechanism which allows tools and other
+controllers to collect summary information about resources without needing to
+understand resource-specific status details. Conditions should complement more
+detailed information about the observed status of an object written by a
+controller, rather than replace it. For example, the "Available" condition of a
```

Comment:
So this really inverts the previous guidance, which was sort of agonized over (and with which I happen to agree). Available (or Ready in the previous edit) are omnipresent and (I think?) are never "unknown".

Why are those correct as Conditions rather than fields?  What principal guides one to decide between those?

Reply:
If I have only one argument to make - it's this one.  As written, many status fields should just become conditions, and I don't think that's right.

Reply:
If we were going to have a distinct subresource for conditions, which we could RBAC distinctly, then I could see an argument forming around it.  That doesn't answer the question for "core" conditions like ready.

Reply:
I agree with Tim.

Reply:
I'd need to look at those more carefully before I could agree that Deployment's conditions are a good example.

Reply:
None of this speaks to the question of principle - what is the litmus test that helps me decide - field or condition?

Reply:
I'd propose the following guidance:

1. Expose the underlying measurements used to make decisions as fields.
1. If the outcome of that decision is interesting to a human (roughly: does it affect the behavior or functionality of the object in an externally observable way), provide a _single_ Condition which summarizes whether that aspect of the controller algorithm is operating correctly ("within the envelope").

Reply:
Conditions summarize characteristics of object lifecycle that must be generic across multiple types, have a need to convey both state and human message, and can be extended by third parties to add more nuance to existing lifecycle.  

An object in Kube is a distributed state machine.  The state specified to that object should always be fields.  Some objects have consistent fields that describe status of the state machine.  One of those is observedGeneration.  Another one is conditions.  Some conditions define generic, multi-object state machine characteristics like Available, Progressing, Ready.